### PR TITLE
chore: remove address prop and update sponsor txn check

### PR DESCRIPTION
**Pull request by abcrane123**
**What changed? Why?**
- remove address prop and use wagmi useAccount instead
- update life cycle status check for transaction sponsor

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: site/docs/pages/transaction/types.mdx [hunk 1]
```diff
@@ -27,7 +27,7 @@ type TransactionError = {
 
 ```ts
 type TransactionReact = {
-  address: Address; // The wallet address involved in the transaction.
+  address?: Address; // The wallet address involved in the transaction (defaults to connected account if not provided).
```
Comment (Zizzamia) [inline site/docs/pages/transaction/types.mdx:30]: And in the end, we decided to remove this completelly, right?

Comment (abcrane123) [reply site/docs/pages/transaction/types.mdx]: yep just updated

File: .changeset/brave-icons-fix.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@coinbase/onchainkit": patch
+---
+
+- **patch**: Remove address prop and fix sponsor component. By @abcrane123 #1114
```
Comment (Zizzamia) [inline .changeset/brave-icons-fix.md:5]: nit. try to be more verbose and explain this as someone with no context is reading this.

File: src/transaction/components/TransactionButton.tsx [hunk 3]
```diff
@@ -26,6 +25,8 @@ export function TransactionButton({
     transactionId,
   } = useTransactionContext();
 
+  const { address } = useAccount();
```
Comment (Zizzamia) [inline src/transaction/components/TransactionButton.tsx:28]: Should `useAccount` be added even in small components. Or should be synced at Provider level?

Open question

File: src/transaction/components/TransactionProvider.test.tsx [hunk 4]
```diff
@@ -440,6 +440,38 @@ describe('TransactionProvider', () => {
...
+      .mockRejectedValue(new Error(METHOD_NOT_SUPPORTED_ERROR_SUBSTRING));
+    const writeContractAsyncMock = vi
+      .fn()
+      .mockRejectedValue({ cause: { name: 'UserRejectedRequestError' } });
+    (useWriteContracts as ReturnType<typeof vi.fn>).mockReturnValue({
+      status: 'idle',
+      writeContractsAsync: writeContractsAsyncMock,
+    });
+    (useWriteContract as ReturnType<typeof vi.fn>).mockReturnValue({
+      status: 'IDLE',
```
Comment (Zizzamia) [inline src/transaction/components/TransactionProvider.test.tsx:456]: idle or IDLE?

I have a feeling it's always `lowercase`

File: src/transaction/components/TransactionSponsor.test.tsx [hunk 5]
```diff
@@ -20,11 +20,30 @@ describe('TransactionSponsor', () => {
...
     (useTransactionContext as vi.Mock).mockReturnValue({
-      lifeCycleStatus: { statusName: 'transactionIdle', statusData: null },
+      lifeCycleStatus: { statusName: 'init', statusData: null },
       hasPaymaster: false,
     });
     render(<TransactionSponsor />);
 
     expect(screen.queryByText('Zero transaction fee')).not.toBeInTheDocument();
   });
+  it('does not render if statusName is not init', () => {
```
Comment (Zizzamia) [inline src/transaction/components/TransactionSponsor.test.tsx:30]: My take on spaces with test is:
- no space within an `it`
- one space between `it`

By doing that it's much easiser reading large file full of tests.

File: src/transaction/components/TransactionSponsor.test.tsx [hunk 6]
```diff
@@ -20,11 +20,30 @@ describe('TransactionSponsor', () => {
...
+    (useTransactionContext as vi.Mock).mockReturnValue({
+      lifeCycleStatus: { statusName: 'blah', statusData: null },
+      hasPaymaster: false,
+    });
+    render(<TransactionSponsor />);
+
+    expect(screen.queryByText('Zero transaction fee')).not.toBeInTheDocument();
+  });
+  it('does render if statusName is init', () => {
```
Comment (Zizzamia) [inline src/transaction/components/TransactionSponsor.test.tsx:39]: Usually the good practice of test, it's to start with the wording `should ...`

File: src/transaction/components/TransactionProvider.test.tsx (hunk 4 above)
Comment (abcrane123) [reply src/transaction/components/TransactionProvider.test.tsx:456]: yeah you're right it is `idle`

File: src/transaction/components/TransactionButton.tsx (hunk 3 above)
Comment (Zizzamia) [reply src/transaction/components/TransactionButton.tsx:28]: let's go ahead with your intuition.

Comment (Zizzamia) [reply src/transaction/components/TransactionButton.tsx:28]: sounds good

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**